src/js/components/InfiniteScroll/InfiniteScroll.js

```diff
@@ -101,21 +105,43 @@ const InfiniteScroll = ({
 
       // Figure out which pages we should make visible based on the scroll
       // window.
+
+      // Scroll window buffers allow for triggering effects ahead of a
+      // user's current scroll position and direction
       const offset = height / 4;
+      const scrollBufferTop = top - offset;
+      const scrollBufferBottom = top + height + offset;
+
+      // Account for page steps containing items of varying heights.
+      const avgPageHeight =
+        pageHeights.reduce((totalPageHeights, page) => {
+          return totalPageHeights + page;
+        }, pageHeight) /
+        (pageHeights.length + 1);
+      const avgPageArea =
+        pageAreas.reduce((totalPageAreas, page) => {
+          return totalPageAreas + page;
+        }, pageArea) /
+        (pageAreas.length + 1);
 
       // nextBeginPage will increment/decrement when using replace, otherwise
       // the beginPage will be at 0.
-      const nextBeginPage = replace
-        ? Math.min(
-            lastPage,
-            Math.max(
-              0,
-              multiColumn
-                ? Math.floor((Math.max(0, top - offset) * width) / pageArea)
-                : Math.floor(Math.max(0, top - offset) / pageHeight),
-            ),
-          )
-        : 0;
+      // If onMore is present nextBeginPage should be 0,
+      // as there is no way of knowing lastPage.
```

Comment:
I think `lastPage` was meant to be the last page of the `data` we have, not the eventual last page when onMore is no longer provided.

Reply:
The main reasoning behind this is to prevent some issues with variable heights when onmore and replace are present.

This is when onmore and replace are present:

(nextEndPage fluctuates between 0 and 1)
![Kapture 2020-09-24 at 10 38 52](https://user-images.githubusercontent.com/21990914/94167719-7318cd00-fe52-11ea-8ff9-4eafe264c85d.gif)

(bottom marker is too tall and causes scrollbar to snap)
![Kapture 2020-09-24 at 10 43 49](https://user-images.githubusercontent.com/21990914/94168249-0ce07a00-fe53-11ea-8240-bdb01b7cb5d5.gif)